src/js/media-captions-menu-button.js

```diff
@@ -0,0 +1,184 @@
+   * if captions selected, the disabled icon should be hidden
+   */
+  media-chrome-menu-button:not(.enabled) .enabled,
+  media-chrome-menu-button.enabled .disabled {
+    display: none;
+  }
+  /*
+   * if captions selected, enabled icon should be shown
+   * if no captions selected, disabled icon should be shown
+   */
+  media-chrome-menu-button.enabled .enabled {
+  media-chrome-menu-button:not(.enabled) .disabled {
+    display: initial;
+  }
+</style>
+
+<media-chrome-menu-button aria-label="captions menu button">
+  ${ccEnabledIcon}
+  ${ccDisabledIcon}
+  <media-captions-listbox slot="listbox" media-controller="mc"></media-captions-listbox>
+</media-chrome-menu-button>
+`;
+
+/**
+ * @extends {HTMLElement}
+ */
+class MediaCaptionsMenuButton extends window.HTMLElement {
+  #changeCallback;
+  #menuButton;
+  #handleClick;
+
+  static get observedAttributes() {
+    return [
+      'disabled',
+      MediaStateReceiverAttributes.MEDIA_CONTROLLER,
+      MediaUIAttributes.MEDIA_CAPTIONS_SHOWING,
+      MediaUIAttributes.MEDIA_SUBTITLES_SHOWING,
+    ];
+  }
+
+  constructor() {
+    super();
+
+    const shadow = this.attachShadow({ mode: 'open' });
+    const mediaCaptionsMenuButton = template.content.cloneNode(true);
+    this.nativeEl = mediaCaptionsMenuButton;
+
+    shadow.append(mediaCaptionsMenuButton);
+
+    // this.#changeCallback = this.#changeCallback_.bind(this);
+    this.#menuButton = this.shadowRoot.querySelector('media-chrome-menu-button');
+
+    this.#handleClick = this.#handleClick_.bind(this);
+  }
+
+  attributeChangedCallback(attrName, oldValue, newValue) {
+    if (attrName === MediaStateReceiverAttributes.MEDIA_CONTROLLER) {
+      if (oldValue) {
+        const mediaControllerEl = document.getElementById(oldValue);
+        mediaControllerEl?.unassociateElement?.(this);
+      }
+      if (newValue) {
+        const mediaControllerEl = document.getElementById(newValue);
+        mediaControllerEl?.associateElement?.(this);
+      }
+    } else if (
+      attrName === MediaUIAttributes.MEDIA_CAPTIONS_SHOWING ||
+      attrName === MediaUIAttributes.MEDIA_SUBTITLES_SHOWING &&
+      newValue !== oldValue
+    ) {
+      if (newValue) {
+        this.#captionsEnabled();
+      } else {
+        this.#captionsDisabled();
+      }
+    } else if (attrName === 'disabled' && newValue !== oldValue) {
+      if (newValue == null) {
+        this.enable();
+      } else {
+        this.disable();
+      }
+    }
+  }
+
+
+  #handleClick_() {
+    this.#updateMenuPosition();
+  }
+
+  #updateMenuPosition() {
+    const svgs = this.shadowRoot.querySelectorAll('svg');
+    const listbox = this.#menuButton.querySelector('media-captions-listbox');
+    const xOn = svgs[0].getBoundingClientRect().x;
+    const xOff = svgs[1].getBoundingClientRect().x;
+    const leftOffset = xOn > 0 ? xOn : xOff > 0 ? xOff : 0;
+    const xBody = /** @type {Document} */(document).body.getBoundingClientRect().x;
```

Comment:
(blocking): doesn't this need to be the bounding box of the container or wrapper?

Reply:
you're right. Maybe `this.parentElement.getBoundingClientRect().x` or something like that should be done instead (with the appropriate null checks).

Reply:
we did a similar thing for the time range preview thumbs. parent element is probably sufficient as first pass.
https://github.com/muxinc/media-chrome/blob/5c7acfbdd0998f6b18e8dc2c01006abfa8ce7b58/src/js/media-time-range.js#L353-L357

Reply:
Ah, nice, I'll take a look there.